### ISSUE TITLE
Error : LinearAlgebra.LAPACKException(3)

### ISSUE THREAD

**Issue by dlecourstessier** [Circuitscape/Omniscape.jl#9]
Hi all !

I am currently running Omniscape, but I get this error :

ERROR: LinearAlgebra.LAPACKException(3)

Stacktrace: [1] chklapackerror at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/lapack.jl:38 [inlined] [2] gesdd!(::Char, ::Array{Float64,2}) at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/lapack.jl:1602 [3] #svd!#66(::Bool, ::Function, ::Array{Float64,2}) at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/svd.jl:63 [4] #svd! at ./none:0 [inlined] [5] #svd#67 at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/svd.jl:106 [inlined] [6] #svd at ./none:0 [inlined] [7] #pinv#31(::Float64, ::Float64, ::Function, ::Array{Float64,2}) at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/dense.jl:1307 [8] pinv at /builddir/build/BUILD/julia/build/usr/share/julia/stdlib/v1.1/LinearAlgebra/src/dense.jl:1286 [inlined] [9] Type at /home/dlecourstessier/.julia/packages/AlgebraicMultigrid/eCNmE/src/multilevel.jl:57 [inlined] [10] AlgebraicMultigrid.Pinv(::SparseArrays.SparseMatrixCSC{Float64,Int64}) at /home/dlecourstessier/.julia/packages/AlgebraicMultigrid/eCNmE/src/multilevel.jl:59 [11] smoothed_aggregation(::SparseArrays.SparseMatrixCSC{Float64,Int64}, ::Type{Val{1}}, ::AlgebraicMultigrid.HermitianSymmetry, ::AlgebraicMultigrid.SymmetricStrength{Float64}, ::AlgebraicMultigrid.StandardAggregation, ::AlgebraicMultigrid.JacobiProlongation{Float64}, ::AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, ::AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, ::AlgebraicMultigrid.GaussSeidel{AlgebraicMultigrid.SymmetricSweep}, ::Int64, ::Int64, ::Bool, ::Bool, ::Type) at /home/dlecourstessier/.julia/packages/AlgebraicMultigrid/eCNmE/src/aggregation.jl:54 [12] smoothed_aggregation at /home/dlecourstessier/.julia/packages/AlgebraicMultigrid/eCNmE/src/aggregation.jl:16 [inlined] (repeats 2 times) [13] macro expansion at ./util.jl:213 [inlined] [14] multiple_solver(::Dict{String,String}, ::SparseArrays.SparseMatrixCSC{Float64,Int64}, ::Array{Float64,1}, ::Array{Float64,1}, ::Array{Float64,1}) at /home/dlecourstessier/.julia/packages/Circuitscape/mKAhh/src/raster/advanced.jl:281 [15] calculate_current(::Array{Float64,2}, ::Array{Float64,2}, ::Array{Float64,2}, ::Circuitscape.RasterFlags, ::Dict{String,String}) at /home/dlecourstessier/.julia/packages/Omniscape/3uB6x/src/functions.jl:265 [16] solve_target!(::Int64, ::Int64, ::Dict{String,Int64}, ::Array{Float64,2}, ::Array{Float64,2}, ::Array{Float64,2}, ::Dict{String,String}, ::Circuitscape.OutputFlags, ::Bool, ::Bool, ::Array{Float64,2}, ::Array{Float64,2}, ::Array{Float64,2}) at /home/dlecourstessier/.julia/packages/Omniscape/3uB6x/src/functions.jl:342 [17] run_omniscape(::String) at /home/dlecourstessier/.julia/packages/Omniscape/3uB6x/src/run_omniscape.jl:145 [18] top-level scope at none:0

Does anyone have an idea how I can fix the problem ?

Thanks a lot !

**Comment by ranjanan** [comment]
Could you give me steps on how to reproduce this error?

**Comment by dlecourstessier** [comment]
Hi !

1) I've created the resistance file (between 0 and 100)
2) I've created the source file (between 75 and 100) 
    --> Both in asc format
3) I've write the initiation file with the .ini denomination
4) I run Omniscape with two cores

Here is a drive document with my resistance and source files, my script and the initiation files
https://drive.google.com/drive/folders/1VHlBAKU2ZlTSQt23JZmJQ9pfd7rOKXaa?usp=sharing

Thanks a lot !

**Comment by vlandau** [comment]
I'll start by taking a look at the inputs, @ranjanan. @dlecourstessier, I would suggest also starting with much smaller resistance and source grids (fewer pixels). A resistance grid of 1GB and a block size of 5 could take weeks to finish running.

**Comment by dlecourstessier** [comment]
Would you recommend upgrading the block size or doing tile ?

**Comment by vlandau** [comment]
For troubleshooting purposes, you might start with a smaller subset of the study area. For the larger run, you could consider running on a computer with many more cores, and/or increase the block size to reduce total compute time. 

I'm also noticing that your `radius` is 50 pixels, which is only 1.5km with a pixel size of 30 meters. Are you looking at connectivity for a small animal or for plant dispersal? Typically, you would want to choose a `radius` the corresponds to the dispersal distance of the study organism. 

One thing I noticed off the bat is that you have source strengths of 75 in areas where there is no data for resistance. This may be causing a bug, but I'm not sure. I'm going to run a few tests and I'll get back to you.

**Comment by vlandau** [comment]
Now I'm seeing that you set the source threshold to 100, which should take care of the 75 values. I will get back to you shortly.

**Comment by vlandau** [comment]
I'm seeing in your input file that the nodata value is set to 1.7976931348623157081e+308, but it needs to be -9999 at this time. This could be what is causing the error.

**Comment by ranjanan** [comment]
@dlecourstessier just curious, why are you using 1.79e308 (the largest possible floating point value representable with 64-bit floating point) as your nodata value?

**Comment by vlandau** [comment]
@ranjanan It's not a user-defined thing. It was probably set to that by default by whatever GIS software that was used. Issue is solved, by the way. Will post back in a minute with a link to the fixed resistance file.

**Comment by dlecourstessier** [comment]
For the 1.79.... no data value, I gest the -9999 "no data value" have been lost in the translation. I can change it without problems ! I'm going to chance it in the google drive doc !
And I am going to change the radius.... 

I will get back to you shortly

**Comment by vlandau** [comment]
@dlecourstessier I'm uploading a fixed resistance file right now. Here it is: https://drive.google.com/open?id=1GkoYysKvc9WTjl2FhXmYp-fg40q1dThy

**Comment by vlandau** [comment]
Okay, this issue is resolved on the master branch with commit [799b8d1](https://github.com/Circuitscape/Omniscape.jl/commit/799b8d18f33b91ae915c591cd579232eca25bf56).

I'm going to wait on a new release with this patch for now in case additional issues come up. For now, you can update Omniscape using the following code in Julia.
```julia
using Pkg; Pkg.add(PackageSpec(url="https://github.com/Circuitscape/Omniscape.jl", rev="master")
```

**Comment by dlecourstessier** [comment]
I've downloaded the file, thanks !
